src/vm/builtins/builtin_runner/poseidon.zig

```diff
@@ -54,37 +69,362 @@ pub const PoseidonBuiltinRunner = struct {
         included: bool,
     ) Self {
         return .{
-            .base = 0,
             .ratio = ratio,
-            .cell_per_instance = poseidon_instance_def.CELLS_PER_POSEIDON,
-            .n_input_cells = poseidon_instance_def.INPUT_CELLS_PER_POSEIDON,
-            .stop_ptr = null,
             .included = included,
             .cache = AutoHashMap(Relocatable, Felt252).init(allocator),
-            .instances_per_component = 1,
         };
     }
 
-    pub fn initSegments(self: *Self, segments: *MemorySegmentManager)  !void {
-        _ = self;
-        _ = segments;
+    pub fn initSegments(self: *Self, segments: *MemorySegmentManager) !void {
+        self.base = @intCast((try segments.addSegment()).segment_index);
     }
 
     pub fn initialStack(self: *Self, allocator: Allocator) !ArrayList(MaybeRelocatable) {
-        _ = self;
         var result = ArrayList(MaybeRelocatable).init(allocator);
-        errdefer result.deinit();        
+        if (self.included) {
+            try result.append(MaybeRelocatable.fromSegment(
+                @intCast(self.base),
+                0,
+            ));
+            return result;
```

Comment:
You can remove this return as long as you also return `result` after your if statement no?